### ISSUE TITLE
Cannot read property initialState of undefined

### ISSUE THREAD

**Issue by Blacktiger** [davidkpiano/xstate#430]
## Bug or feature request?
Probably more of a feature request, but since I'm providing reproduction steps maybe it's really a bug?

## Description:

Given the following example state machine I get an error which doesn't make it obvious what the problem with my machine is. I believe it's just because I'm invoking a service that doesn't return a promise but it's unclear from the error message what is wrong.

Machine
```
const formMachine = Machine({
  id: 'form',
  initial: 'valid',

  states: {
    valid: {
      on: {
        SAVE: 'pending'
      }
    },

    pending: {
      invoke: {
        src: 'onSubmit',
        onDone: {
          target: 'success'
        }
      }
    },

    success: {
      type: 'final',
      actions: 'onDone'
    }
  }
},
{
  actions: {
    onDone: (context, event) => console.log('done', context, event)
  },

  services: {
    onSubmit: (context, event) => console.log('submit', context, event)
  }
});
```

## (Bug) Expected result:

Ideally this should return an error message like "service does not return a promise, callback function, or machine".

## (Bug) Actual result:

```
interpreter.js:354 Uncaught TypeError: Cannot read property 'initialState' of undefined
    at t.start (interpreter.js:354)
    at t.spawn (interpreter.js:622)
    at t.exec (interpreter.js:549)
    at interpreter.js:227
    at Array.forEach (<anonymous>)
    at t.execute (interpreter.js:226)
    at t.update (interpreter.js:238)
    at t.flush (interpreter.js:417)
    at StateChart.send (interpreter.js:150)
    at onClick (StateChartNode.js:146)
```

## (Bug) Potential fix:

When calling a service that we expect to return a promise, the library should throw an error when it sees that the service doesn't return a promise. It looks like the failure is around here: https://github.com/davidkpiano/xstate/blob/320166e426f549776dc7811cb6b3d11f84f202b5/src/interpreter.ts#L690 The logic there detects no promise or callback function so it seems to assume the service is a machine when in actuality the library user (me) made a stupid error from not understanding how xstate works. Instead a check should be made to determine if the service actually returned a machine and if not it should just throw an error to help the user fix the problem.

**Comment by davidkpiano** [comment]
One way to avoid this problem is to use [TypeScript](http://typescriptlang.org), but of course, this might not be feasible for every developer/team. But yes, there should be better checks to determine that it is actually a machine.

Workaround for now:

```js
services: {
  onSubmit: (ctx, e) => () => console.log('submit', ctx, e)
}
```

**Comment by Blacktiger** [comment]
Yeah, once I figured out what I was doing wrong I was able to fix the problem. As you showed, simply adding a callback function there will do what I wanted. This was the latest error I bumped into, but I do feel like handling incorrect statechart definitions could use some tightening up from a validation perspective since there were other cases where it took me a bit to determine what I had done wrong.
